app/lib/image_resizer.rb

```diff
@@ -0,0 +1,22 @@
+module ImageResizer
+  DEFAULT_CL_OPTIONS = {
+    type: "fetch",
+    height: nil,
+    width: nil,
+    crop: "limit",
+    quality: "auto",
+    flags: "progressive",
+    fetch_format: "auto",
+    sign_url: true
+  }.freeze
+
+  def self.call(img_src, **kwargs)
+    options = DEFAULT_CL_OPTIONS.merge(kwargs).reject { |_, v| v.blank? }
```

Comment:
Unless you are worried about empty strings, you can replace `.reject { |_, v| v.blank? }` with `compact`:

```ruby
{ width: nil, height: nil, crop: "limit" }.compact
#=> {:crop=>"limit"}
```

But this also raises the question: why include the `nil` options in `DEFAULT_CL_OPTIONS` when we then remove them here? Is it mostly for documentation purposes so people know what they can pass in?

Reply:
I haven't thoroughly tested this out but based on Cloudinary's documentation, depending on the `crop` option given, you don't need to specify both width and height. I'm definitely not worried about empty strings so I will change it to `compact`.